vmware/terraform-provider-tanzu-mission-control
# bug: fix for eks cluster and nodepool tag issue

### Summary
Change in the order in which cluster and nodepool are updated.
This PR fixes the issue of nodepool tags not getting updated after the tags are removed from cluster and added to nodepool in the same resource file.


### Type



- [x] This is a bug fix.
- [ ] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

### Breaking Changes?



- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation



- [x] Tests have been completed.



### Issue References



### Additional Information
Verification :-
1. Add a tag in the cluster. Now once the cluster tags are updated, remove the tag from cluster spec by using **tags = { }** and add it to the nodepool spec. The tag is removed from the cluster and added in the nodepool spec now.
2. Add/Remove tags from nodepool works as expected.
3. Add/Remove tags from cluster works as expected.
